src/init/AppRunner.ts

```diff
@@ -33,41 +35,65 @@ export class AppRunner {
   /**
    * Starts the server with a given config.
    * This method can be used to start the server from within another JavaScript application.
+   *
```

Comment:
My understanding of this module became harder and harder because of:
- settings
- cli
- cliSettings
- variable settings

It's not a nit; I really lost track of understanding.

Can we have clear names for things like `cliDefaults`, `cliValues` or what exactly it is that they are?

Reply:
Even `variableBindings` is confusing here; are there parameter bindings, too?

Maybe there `configVariables` or `variableValues` or `variableDefaults`.

Happy to help us think, I'd just need an enumeration of what kinds of data we are dealing with.

Reply:
The names/types are defined in https://github.com/CommunitySolidServer/CommunitySolidServer/blob/c902638bde175768defa81ef5381831a3c1d1d2a/src/init/variables/Types.ts#L1-L16

2 ways to start the server

Start through CLI:
 * Receives `CliArgv` as input (just noticed this type is not used consistently and sometimes still `argv: string[]` is used).
 * Converts `CliArgv` to `Settings` (a JS object corresponding to parsed CLI values, so `['--file', './data']` becomes `{ file: './data' }`). 
 * Converts `Settings` to `VariableBindings` (the object we will give to Components.js to start the server, so `{ 'urn:solid-server:default:variable:rootFilePath': './data' }`). This is the step where the default values for the settings will be applied.
 * Starts the server with those generated bindings.

Start through code:
 * Potentially receives both `Settings` and `VariableBindings` objects as input.
 * Converts `Settings` to `VariableBindings` (this part is new in this PR, causes the default values to also be applied when starting through code).
 * Merges the generated and received `VariableBindings`.
 * Starts the server with those bindings.

Reply:
Check, thanks! Maybe the confusing name is `settings` then, as discussed above.